### PR TITLE
P4-16 Eliminate requirement that table action bodies must be atomic

### PR DESCRIPTION
**Pull request by jafingerhut**

### CODE REVIEW

**Comment by jafingerhut** [comment]
At most one of this PR and https://github.com/p4lang/p4-spec/pull/559 should be merged.  They are somewhat different proposals for the P4 language working group to consider.

File: p4-16/spec/P4-16-spec.mdk [hunk 1]
```diff
@@ -6201,6 +6197,12 @@ in actions invoked from tables `flowlet` (reading) and `new_flowlet`
 would not execute atomically: a second packet may read the state of `r`
 before the first packet had a chance to update it.
 
+Note that even within an `action` definition, if the action does
+something like reading a register, modifying it, and writing it back,
+in a way that only the modified value should be visible to the next
+packet, then that portion of the action definition should be enclosed
```
Comment (cc10512) [inline p4-16/spec/P4-16-spec.mdk]: ... then, to guarantee correct execution in all cases, that portion ...

Comment (jafingerhut) [reply p4-16/spec/P4-16-spec.mdk]: Done.

**Comment by cc10512** [comment]
@jnfoster do you mind taking a look at this PR and merge if you consider it ready?

**Comment by jafingerhut** [comment]
In 2018-Feb-12 P4 language WG meeting, we decided to merge this change, and not the alternate proposal in PR #559, which will soon be closed without being merged.

**Pull request merged**